mutiny-core/src/hermes.rs

```diff
@@ -338,8 +419,77 @@ impl<S: MutinyStorage> HermesClient<S> {
             None => None,
         }
     }
+}
 
-    // TODO need a way to change the federation if the user's federation changes
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ChangeFederationRequest {
+    pub name: Option<String>,
+    pub federation_id: Option<FederationId>,
+}
+
+async fn change_federation_info(
+    http_client: &reqwest::Client,
+    base_url: &str,
+    nostr_client: Client,
+    current_address: Arc<RwLock<(Option<String>, bool)>>,
+    federation: FederationIdentity,
+    logger: &MutinyLogger,
+) -> Result<(), MutinyError> {
+    // make sure name is registered already
+    if current_address.read().await.0.is_none() {
+        return Ok(());
+    }
```

Comment:
This and the one for `disable_zaps` has a minor race condition if the user manages to remove/switch federation before the check call has finished. Could do rechecking until `.1` is true